src/EditorFeatures/Core/ExternalAccess/IntelliCode/IntentProcessor.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            return lazyIntentProviders.ToImmutableDictionary(
+                provider => (provider.Metadata.LanguageName, provider.Metadata.IntentName),
+                provider => provider);
+        }
+
+        public async Task<ImmutableArray<IntentResult>> ComputeEditsAsync(IntentRequestContext intentRequestContext, CancellationToken cancellationToken)
+        {
+            var currentDocument = intentRequestContext.CurrentSnapshotSpan.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
+            Contract.ThrowIfNull(currentDocument);
+
+            var currentText = await currentDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var originalDocument = currentDocument.WithText(currentText.WithChanges(intentRequestContext.PriorTextEdits));
+
+            var languageName = originalDocument.Project.Language;
+            if (!_lazyIntentProviders.TryGetValue((LanguageName: languageName, IntentName: intentRequestContext.IntentName), out var provider))
+            {
+                Logger.Log(FunctionId.Intellicode_UnknownIntent, KeyValueLogMessage.Create(LogType.UserAction, m =>
+                {
+                    m["intent"] = intentRequestContext.IntentName;
+                    m["language"] = languageName;
+                }));
+
+                return ImmutableArray<IntentResult>.Empty;
+            }
+
+            var selectionTextSpan = intentRequestContext.PriorSelection;
+            var results = await provider.Value.ComputeIntentAsync(
+                originalDocument,
+                selectionTextSpan,
+                currentDocument,
+                intentRequestContext.IntentData,
+                cancellationToken).ConfigureAwait(false);
+            if (results.IsDefaultOrEmpty)
+            {
+                return ImmutableArray<IntentResult>.Empty;
+            }
+
+            using var _ = ArrayBuilder<IntentResult>.GetInstance(out var convertedResults);
+            foreach (var result in results)
+            {
+                var convertedIntent = await ConvertToIntelliCodeResultAsync(result, originalDocument, currentDocument, cancellationToken).ConfigureAwait(false);
+                convertedResults.AddIfNotNull(convertedIntent);
+            }
+
+            return convertedResults.ToImmutable();
+        }
+
+        private static async Task<IntentResult?> ConvertToIntelliCodeResultAsync(IntentProcessorResult processorResult, Document originalDocument, Document currentDocument, CancellationToken cancellationToken)
+        {
+            var title = processorResult.Title;
+            var newSolution = processorResult.Solution;
+
+            // Merge linked file changes so all linked files have the same text changes.
+            newSolution = await newSolution.WithMergedLinkedFileChangesAsync(originalDocument.Project.Solution, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
interesting case.  id this actually come up with your test cases?

Reply:
It did not for this particular refactoring - generate ctor will only generate parameters for the current tfm for example.  However, it will definitely come up later if we do any kind of rename.  I could leave it out until then, but I don't think there's any harm in leaving it in - we always would want to return edits based on whatever is actually in the buffer (the merged result)